### PR TITLE
Add default method support to virtual statics

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Corresponds to #64717 that I somehow missed, despite commenting on it.

We still don't allow interfaces to implement methods of other interfaces. The CLR VM doesn't allow either. That's #66887 that I confused #64717 with.

Unblocks #69756.

Cc @dotnet/ilc-contrib

### CODE REVIEW

File: src/tests/nativeaot/SmokeTests/UnitTests/Interfaces.cs [hunk 1]
```diff
@@ -1020,4 +1021,77 @@ public static void Run()
             TestVariantInterface<GenericVariantWithHiddenDerived<Derived>, Mid>("GenericVariantWithHiddenBase.WhichMethod(Mid)");
         }
     }
+
+    class TestSimpleStaticDefaultInterfaceMethods
+    {
+        interface IFoo
+        {
+            static virtual string GetCookie() => nameof(IFoo);
```
Comment (tannergooding) [inline src/tests/nativeaot/SmokeTests/UnitTests/Interfaces.cs:1029]: This is going to need https://github.com/dotnet/runtime/pull/69756/commits/3e7142334eb128c75b01da7885c11949ec9cc65a

That is you need `<MicrosoftNetCompilersToolsetVersion>4.3.0-2.22270.4</MicrosoftNetCompilersToolsetVersion>` or later

File: src/coreclr/tools/Common/Compiler/TypeExtensions.cs [hunk 2]
```diff
@@ -562,6 +562,16 @@ public static MethodDesc TryResolveConstraintMethodApprox(this TypeDesc constrai
                 method = null;
             }
 
+            // Default implementation logic, which only kicks in for default implementations when lookin up on an exact interface target
```
Comment (jkotas) [inline src/coreclr/tools/Common/Compiler/TypeExtensions.cs:565]:
```suggestion
            // Default implementation logic, which only kicks in for default implementations when looking up on an exact interface target
```

**Review by jkotas — APPROVED** [review]

**Comment by trylek** [comment]
CF https://github.com/dotnet/runtime/issues/69829

**Comment by jkoritzinsky** [comment]
There's an ongoing issue tracked at #69829 .

**Pull request merged**